src/sql/sqlRestApiClient.ts

```diff
@@ -73,17 +75,27 @@ export default class SqlRestApiClient extends SqlApiClient {
    * Runs the given SQL query on the backend.
    * @param {string} query - The raw SQL query to run.
    * @returns {Promise<SqlQueryResult>} - A structured query result from running the raw SQL query.
+   * @throws {MindsDbError} - Something went wrong sending the API request.
    */
   override async runQuery(query: string): Promise<SqlQueryResult> {
     const queryRequest = {
       query,
     };
-    const queryResponse = await this.client.post(
-      this.getQueryUrl(),
-      queryRequest,
-      getBaseRequestConfig(this.session)
-    );
-    const responseData: SqlApiResponse = queryResponse.data;
-    return this.makeQueryResult(responseData);
+    try {
+      const queryResponse = await this.client.post(
```

Comment:
Just in my minds, I think a encapsulation(destructuring) with the 'this' will be more readeable, could be a fool mention. 

`const { getQueryUrl, session,  client } = this;`

In some cases, the 'this' could be vulnerable to side effects